gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java

```diff
@@ -169,6 +172,24 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
         .subscribe(args -> showAddFeatureLayerSelector(args.first, args.second));
   }
 
+  private void onPolygonDrawingStateUpdated(PolygonDrawingState state) {
+    Timber.v("PolygonDrawing state : %s", state);
+    switch (state.getState()) {
+      case IN_PROGRESS:
+        mapContainerViewModel.setViewMode(Mode.DRAW_POLYGON);
+        break;
+      case CANCELED:
+        mapContainerFragment.setDefaultMode();
```

Comment:
While you're at it would you mind changing this api to something more idiomatic? In Java I'd expect to see something like `setMode(DEFAULT)`, `setMode(DRAW_POLYGON)`, `setMode(MOVE_POINT)`, etc.

Reply:
Done